libgraph/src/analytics/k_shortest_paths/sssp.cpp

```diff
@@ -0,0 +1,462 @@
+  using BFSReqPushWrap = typename BFS::ReqPushWrap;
+  using BFSOutEdgeRangeFn = typename BFS::OutEdgeRangeFn;
+  using BFSTileRangeFn = typename BFS::TileRangeFn;
+
+  class PathAlloc {
+  public:
+    Path* NewPath() {
+      Path* path = allocator_.allocate(1);
+      allocator_.construct(path, Path());
+      return path;
+    }
+
+    void DeletePath(Path* path) {
+      allocator_.destroy(path);
+      allocator_.deallocate(path, 1);
+    }
+
+  private:
+    katana::FixedSizeAllocator<Path> allocator_;
+  };
+
+  template <typename Item, typename PushWrap, typename EdgeRange>
+  bool CheckReachabilityAsync(
+      GraphTy* graph, const GNode& source, const PushWrap& pushWrap,
+      const EdgeRange& edgeRange, unsigned int reportNode) {
+    using FIFO = katana::PerSocketChunkFIFO<kChunkSize>;
+    using WL = FIFO;
+
+    using Loop = katana::ForEach;
+
+    Loop loop;
+
+    graph->template GetData<NodeCount>(source) = 1;
+    katana::InsertBag<Item> initBag;
+
+    pushWrap(initBag, source, 1, "parallel");
+
+    loop(
+        katana::iterate(initBag),
+        [&](const Item& item, auto& ctx) {
+          for (auto ii : edgeRange(item)) {
+            GNode dst = graph->OutEdgeDst(ii);
+            if (graph->template GetData<NodeCount>(dst) == 0) {
+              graph->template GetData<NodeCount>(dst) = 1;
+              pushWrap(ctx, dst, 1);
+            }
+          }
+        },
+        katana::wl<WL>(), katana::loopname("runBFS"),
+        katana::disable_conflict_detection());
+
+    if (graph->template GetData<NodeCount>(reportNode) == 0) {
+      return false;
+    }
+
+    katana::do_all(katana::iterate(*graph), [&graph](GNode n) {
+      graph->template GetData<NodeCount>(n) = 0;
+    });
+
+    return true;
+  }
+
+  bool CheckReachabilitySync(
+      GraphTy* graph, const GNode& source, unsigned int reportNode) {
+    katana::InsertBag<GNode> current_bag;
+    katana::InsertBag<GNode> next_bag;
+
+    current_bag.push(source);
+    graph->template GetData<NodeCount>(source) = 1;
+
+    while (current_bag.begin() != current_bag.end()) {
+      katana::do_all(
+          katana::iterate(current_bag),
+          [&](GNode n) {
+            for (auto edge : graph->OutEdges(n)) {
+              auto dest = graph->OutEdgeDst(edge);
+              if (graph->template GetData<NodeCount>(dest) == 0) {
+                graph->template GetData<NodeCount>(dest) = 1;
+                next_bag.push(dest);
+              }
+            }
+          },
+          katana::steal());
+
+      current_bag.clear();
+      std::swap(current_bag, next_bag);
+    }
+
+    if (graph->template GetData<NodeCount>(reportNode) == 0) {
+      return false;
+    }
+
+    katana::do_all(katana::iterate(*graph), [&graph](GNode n) {
+      graph->template GetData<NodeCount>(n) = 0;
+    });
+
+    return true;
+  }
+
+  //delta stepping implementation for finding a shortest path from source to report node
+  template <
```

Comment:
also see other apps on how to do a dxoygen type documentation

Reply:
Any suggestions on specific apps to look at for doxygen type documentation? These were pulled from the original ksp implementation and sssp file

Reply:
seems like the other apps are behind. No worries. Check this one: https://github.com/uditagar/katana-enterprise/blob/0e20f5bb75b1a304208aa97d701d47b0a93ac24b/libdistgalois/src/dist-analytics/louvain-clustering/LouvainClustering.cpp#L92